modules/lang-painless/src/main/java/org/elasticsearch/painless/DecorateExecutePhase.java

```diff
@@ -0,0 +1,310 @@
+                catchBlockNode.setLocation(internalLocation);
+                catchBlockNode.setAllEscape(true);
+                catchBlockNode.setStatementCount(1);
+
+                catchNode.setBlockNode(catchBlockNode);
+
+                throwNode = new ThrowNode();
+                throwNode.setLocation(internalLocation);
+
+                catchBlockNode.addStatementNode(throwNode);
+
+                unboundCallNode = new UnboundCallNode();
+                unboundCallNode.setLocation(internalLocation);
+                unboundCallNode.setExpressionType(ScriptException.class);
+                unboundCallNode.setLocalFunction(new LocalFunction(
+                                "convertToScriptException",
+                                ScriptException.class,
+                                Arrays.asList(Throwable.class, Map.class),
+                                true,
+                                false
+                        )
+                );
+
+                throwNode.setExpressionNode(unboundCallNode);
+
+                variableNode = new VariableNode();
+                variableNode.setLocation(internalLocation);
+                variableNode.setExpressionType(ScriptException.class);
+                variableNode.setName(name);
+
+                unboundCallNode.addArgumentNode(variableNode);
+
+                callNode = new CallNode();
+                callNode.setLocation(internalLocation);
+                callNode.setExpressionType(Map.class);
+
+                unboundCallNode.addArgumentNode(callNode);
+
+                StaticNode staticNode = new StaticNode();
+                staticNode.setLocation(internalLocation);
+                staticNode.setExpressionType(Collections.class);
+
+                callNode.setLeftNode(staticNode);
+
+                callSubNode = new CallSubNode();
+                callSubNode.setLocation(internalLocation);
+                callSubNode.setExpressionType(Map.class);
+                callSubNode.setBox(Collections.class);
+                callSubNode.setMethod(new PainlessMethod(
+                        Collections.class.getMethod("emptyMap"),
+                        Collections.class,
+                        null,
+                        Collections.emptyList(),
+                        null,
+                        null,
+                        null
+                ));
+
+                callNode.setRightNode(callSubNode);
+            }
+
+            blockNode = new BlockNode();
+            blockNode.setLocation(blockNode.getLocation());
+            blockNode.setAllEscape(blockNode.doAllEscape());
+            blockNode.setStatementCount(blockNode.getStatementCount());
+            blockNode.addStatementNode(tryNode);
+
+            executeFunctionNode.setBlockNode(blockNode);
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    protected DecorateExecutePhase() {
```

Comment:
this can be private, no need to construct right?

Reply:
Fixed.